lib/common/xpath.c

```diff
@@ -297,3 +298,23 @@ xml_get_path(xmlNode *xml)
     }
     return NULL;
 }
+
+char *
+pcmk__xpath_node_id(const char *xpath, const char *node)
+{
+    char *retval = NULL;
+    char *patt = crm_strdup_printf("/%s[@id=", node);
+    char *tmp = strstr(xpath, patt);
+
+    if (tmp) {
+        tmp += strlen(patt);
+        tmp++;
+
+        retval = strdup(tmp);
+        tmp = strstr(retval, "\'");
+        CRM_ASSERT(tmp);
+        tmp[0] = 0;
```

Comment:
Not a big deal, but it would be more efficient to find the quote first and use strndup()

Reply:
Fixed, added some extra tests, and added a test case.